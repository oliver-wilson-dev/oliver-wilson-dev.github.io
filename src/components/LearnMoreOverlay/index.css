@import "../CookieDisclaimer/index.css";

:root {
    font-family: 'Source Sans Pro', sans-serif;
}

.background {
    left: 0;
    right: 0;
    bottom: 0;
    height: 0;
    z-index: 6;
    display: flex;
    position: fixed;
    overflow: hidden;
    background: var(--overlayBackground);
    box-sizing: border-box;
    justify-content: center;
    opacity: 0;
    height: 100%;
    transition: opacity .5s cubic-bezier(.35,.42,.15,1.15) .7s;

    &__loaded {
        opacity: 1;
    }

    &__closed {
        opacity: 0;
    }

    & h2, h3 {
        font-family: 'Source Sans Pro', sans-serif;
    }
}

.crossBtn {
    border: none;
    margin-left: 10px;
    background: transparent;
}

.crossIcon {
    width: 25px;
    height: 25px;
    fill: var(--primaryTextColorLight);
}

.overlayHeader {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background: #F4D03F;
    padding: 20px;
    box-shadow: 0px 0px 4px 1px var(--primaryBorderColourLight);
    font-size: 18px;
}

.information {
    overflow-y: scroll;
    overflow-x: hidden;
    padding: 10px 20px;
}

.information::-webkit-scrollbar {
    display: none;
}

.title {
    margin: 0;
    color: var(--cookie-disclaimer-text-colour);
}

.cookieDescription {
    font-size: 16px;
    margin-bottom: 10px;
    line-height: 2em;
}

.listOfCookies {
    padding-left: 25px;
}

.layout {
    display: flex;
    flex-direction: column;
    max-width: 800px;
    overflow: hidden;
    position: relative;
    background-color: var(--primaryBackgroundLight);
    margin: 20px;
    box-shadow: 0px 0px 2px 1px var(--primaryBorderColourLight)
}

.text {
    color: var(--cookie-disclaimer-text-colour);
    font-size: var(--cookie-disclaimer-text-font-size);
}

.paragraph {
    text-align: justify;
}

@media (min-width: 767px) {
    .information {
        padding: 10px 40px;
    }

    .overlayHeader {
        font-size: 28px;
    }
}