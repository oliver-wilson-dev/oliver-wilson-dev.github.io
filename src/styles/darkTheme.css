@import "./index.css";

[data-theme="DARK_THEME"] {
    --primaryBackground: #333;
    --primaryBackgroundImage: linear-gradient(#333, #333);
    --primaryBorderColour: #f8f8f8;
    --primaryTextColor: #f8f8f8;
    --spinnerTrailColour: rgba(248,248,248, 0.2);
    --sectionFadeColourFrom: rgba(51, 51, 51, 0);
    --sectionFadeColourTo: rgba(51, 51, 51, 1);
    --unvisitedLink: var(--projectGreen)
}