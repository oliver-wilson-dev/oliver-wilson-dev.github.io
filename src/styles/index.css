:root {
    --primaryBackgroundLight: #f8f8f8;
    --primaryBackgroundDark: #333;
    --primaryBackground: var(--primaryBackgroundLight);

    --primaryBorderColour: #0b0a0a;
    --primaryBorderColourLight: #0b0a0a;

    --primaryTextColor: #333;

    --primaryTextColorDark: #f8f8f8;
    --primaryTextColorLight: #333;

    --primaryTextColorInverse: var(--primaryTextColorLight);

    --spinnerTrailColour: rgba(11,10,10, 0.2);
    --sectionFadeColourFrom: rgba(248, 248, 248, 0);
    --sectionFadeColourTo: rgba(248, 248, 248, 1);

    --linkBlue: rgb(21, 64, 204);
    --unvisitedLink: var(--linkBlue);

    --profilePictureBorder: var(--primaryBorderColour);

    --headerHeight: 65px;
    --headerPadding: 15px;

    --overlayBackground: rgba(0,0,0,.288);

    --projectGreen: #06d6a0;
    --projectPink: #EF476F;
    --projectYellow: #FFD166;
    --projectBlue: #118AB2;
    --projectDarkBlue: #073B4C;
}