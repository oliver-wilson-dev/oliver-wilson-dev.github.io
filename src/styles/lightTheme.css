@import "./index.css";

[data-theme="LIGHT_THEME"] {
    --primaryBackground: var(--primaryBackgroundLight);
    --primaryBackgroundInverse: var(--primaryBackgroundDark);

    --primaryBorderColour: #0b0a0a;
    --primaryBorderColourLight: #0b0a0a;

    --primaryTextColor: var(--primaryTextColorLight);
    --primaryTextColorInverse: var(--primaryTextColorDark);

    --spinnerTrailColour: rgba(11,10,10, 0.2);
    --sectionFadeColourFrom: rgba(248, 248, 248, 0);
    --sectionFadeColourTo: rgba(248, 248, 248, 1);

    --primaryBackgroundLight: #f8f8f8;

    --unvisitedLink: var(--projectGreen);
    --unvisitedLinkInverse: var(--projectGreen);
}