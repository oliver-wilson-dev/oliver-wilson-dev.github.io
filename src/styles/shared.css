.bodyFixed {
    position: fixed;
    left: 0;
    width: 100%;
    overflow-y: scroll;
}

.pageHeader {
    font-size: 5em;
    margin: 0;
    margin-top: 10px;
    background-clip: text;
    -webkit-background-clip: text;
    letter-spacing: 2px;
    text-align: center;
    width: 100%;
  }

  @media (min-width: 786px) {
    .landing--header {
    max-width: 800px;
    }
  }

  @media (min-width: 0px) and (max-width: 767px) {
    .pageHeader {
      font-size: 3em;
    }
  }