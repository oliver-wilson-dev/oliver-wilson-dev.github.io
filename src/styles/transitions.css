:root {
    --spinnerOpacityTransitionDuration: 1s;
    --sectionOpacityTransitionDuration: 0.5s;
    --sectionVisibilityTransitionDuration: 0s;
}