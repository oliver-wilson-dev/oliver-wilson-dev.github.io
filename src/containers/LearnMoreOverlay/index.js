import { connect } from 'react-redux';
import LearnMoreOverlay from '../../components/LearnMoreOverlay';
import learnMoreClicked from '../../state/actions/learnMoreClicked';

const mapDispatchToProps = dispatch => ({
  learnMoreClicked: () => dispatch(learnMoreClicked())
});

export default connect(undefined, mapDispatchToProps)(LearnMoreOverlay);
