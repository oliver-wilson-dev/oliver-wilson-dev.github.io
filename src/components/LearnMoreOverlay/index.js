/* eslint-disable max-len */

import React, { useState, useEffect } from 'react';
import cn from 'classnames';
import styles from './index.css';

const LearnMoreOverlay = () => {
  const [acknowledged, setAcknowledged] = useState(false);
  const [loaded, setLoaded] = useState(false);
  const [closed, setClosed] = useState(false);

  useEffect(() => {
    setLoaded(true);
  }, []);

  const onTransitionEnd = () => {
    if (acknowledged) setClosed(true);
  };

  const acknowledgeBanner = () => {
    setAcknowledged(true);
  };

  return (
    <div
      className={cn(styles.container, {
        [styles.container__loaded]: loaded,
        [styles.container__closed]: acknowledged,
      })}

      onTransitionEnd={onTransitionEnd}
    >
      <div className={styles.layout}>
        <div className={cn(styles.information, styles.text)}>
          <h2 className={styles.title}>Cookie Policy</h2>
          <p>This site uses cookies, but you can control these through your browser settings. Most browsers allow you to manage cookies saved on your device â€“ just head to the help section of your browser.</p>
          <h3>What are cookies?</h3>
          <p>Cookies are data files that can hold small amounts of info and are stored on your device (computer, smartphone etc) when you visit a website.</p>
          <h3>What cookies does this site use?</h3>
          <ul className={styles.listOfCookies}>
            <li className={styles.cookieDescription}>This site stores cookie data based on the theme selector if you choose to use it. A cookie is written detailing the theme choice which enables the site to remember your theme selection and preserve that selection when you refresh the page.</li>
            <li className={styles.cookieDescription}>This site uses cookies to record that you&apos;ve acknowledged this cookie policy so that we don&apos;t have to show it to you more than we need to.</li>
          </ul>
        </div>
        <button
          className={cn(styles.cross, {
            [styles.cross__loaded]: loaded && !closed,
            [styles.cross__closed]: acknowledged,
          })}
          type="button"
          onClick={acknowledgeBanner}
        />
      </div>
    </div>
  );
};

export default LearnMoreOverlay;
