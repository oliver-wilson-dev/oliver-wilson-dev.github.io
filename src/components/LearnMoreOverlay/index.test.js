import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import LearnMoreOverlay from '.';
import styles from './index.css';

const defaultProps = {
  learnMoreClicked: jest.fn()
};

const render = (renderMethod = shallow) => renderMethod(<LearnMoreOverlay {...defaultProps} />);

describe('<Landing/>', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });


  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render(mount)).toMatchSnapshot();
  });

  describe('after the first animation', () => {
    it('should render correctly', () => {
      const component = render(mount);

      act(() => {
        component.find(`.${styles.background}`).props().onTransitionEnd();
      });

      component.update();

      expect(component).toMatchSnapshot();
    });
  });

  describe('when clicking the close button', () => {
    it('should call the learnMoreClicked prop', () => {
      const component = render(mount);

      component.find(`.${styles.cross}`).simulate('click');

      let onTransitionEnd;
      act(() => {
        ({ onTransitionEnd } = component.find(`.${styles.background}`).props());
        onTransitionEnd();
      });

      component.update();

      expect(defaultProps.learnMoreClicked).toHaveBeenCalled();
    });

    describe('when the user had not scrolled before opening the modal', () => {
      it('should scroll the user to the scrollY position', () => {
        const component = render(mount);
        const mockScrollY = undefined;
        document.body.style.top = mockScrollY;

        Object.defineProperty(document.body.style, 'top', { value: mockScrollY, writable: true });

        component.find(`.${styles.cross}`).simulate('click');

        act(() => {
          component.find(`.${styles.background}`).props().onTransitionEnd();
        });

        component.update();

        expect(window.scrollTo.mock.calls[0][1]).toBe(-0);
      });
    });

    describe('when the user had scrolled before opening the modal', () => {
      it('should scroll the user to the scrollY position', () => {
        const mockScrollY = 500;
        window.scrollY = mockScrollY;
        const component = render(mount);

        component.find(`.${styles.cross}`).simulate('click');

        act(() => {
          component.find(`.${styles.background}`).props().onTransitionEnd();
        });

        component.update();

        expect(window.scrollTo.mock.calls[0][1]).toBe(mockScrollY);
      });
    });
  });
});
