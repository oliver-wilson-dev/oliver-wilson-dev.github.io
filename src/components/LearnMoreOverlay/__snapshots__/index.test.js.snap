// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing/> after the first animation should render correctly 1`] = `
<LearnMoreOverlay
  learnMoreClicked={[MockFunction]}
>
  <div
    className="background background__loaded"
    onTransitionEnd={[Function]}
  >
    <div
      className="layout"
    >
      <div
        className="overlayHeader"
      >
        <h2
          className="title"
        >
          Cookie Policy
        </h2>
        <button
          className="cross"
          onClick={[Function]}
          type="button"
        />
      </div>
      <div
        className="information text"
      >
        <span>
          Last updated: 6th June 2020
        </span>
        <h3>
          What are cookies?
        </h3>
        <p
          className="paragraph"
        >
          A “cookie” is a string of information which assigns you a unique identifier that we store on your computer. Your browser then provides that unique identifier to use each time you submit a query to the Site. We use cookies on the Site to, among other things, keep track of services you have used, record registration information, record your user preferences, keep you logged into the Site, facilitate purchase procedures, and track the pages you visit. Cookies help us understand how the Site is being used and improve your user experience.
        </p>
        <h3>
          What cookies does this site use?
        </h3>
        <ul
          className="listOfCookies"
        >
          <li
            className="cookieDescription"
          >
            This site stores cookie data based on the theme selector if you choose to use it. A cookie is written detailing the theme choice which enables the site to remember your theme selection and preserve that selection when you refresh the page.
          </li>
          <li
            className="cookieDescription"
          >
            This site uses cookies to record that you've acknowledged this cookie policy so that we don't have to show it to you more than we need to.
          </li>
        </ul>
        <h3>
          Your control of cookies
        </h3>
        <p
          className="paragraph"
        >
          This site uses cookies, but you can control these through your browser settings. Most browsers allow you to manage cookies saved on your device – just head to the help section of your browser.
        </p>
        <p
          className="paragraph"
        >
          Most browsers are set to accept cookies by default. However, you can remove or reject cookies in your browser’s settings. Please be aware that such action could affect the availability and functionality of the Site.
        </p>
        <h3>
          Amendments to this cookie policy
        </h3>
        <p
          className="paragraph"
        >
          We reserve the right to make changes to this Cookie Policy at any time and for any reason. We will alert you about any changes by updating the “Last Updated” date of this Cookie Policy. Any changes or modifications will be effective immediately upon posting the updated Cookie Policy on the Site, and you waive the right to receive specific notice of each such change or modification.
        </p>
        <p
          className="paragraph"
        >
          You are encouraged to periodically review this Cookie Policy to stay informed of updates. You will be deemed to have been made aware of, will be subject to, and will be deemed to have accepted the changes in any revised Cookie Policy by your continued use of the Site after the date such revised Cookie Policy is posted.
        </p>
      </div>
    </div>
  </div>
</LearnMoreOverlay>
`;

exports[`<Landing/> should render correctly 1`] = `
<LearnMoreOverlay
  learnMoreClicked={[MockFunction]}
>
  <div
    className="background background__loaded"
    onTransitionEnd={[Function]}
  >
    <div
      className="layout"
    >
      <div
        className="overlayHeader"
      >
        <h2
          className="title"
        >
          Cookie Policy
        </h2>
        <button
          className="cross"
          onClick={[Function]}
          type="button"
        />
      </div>
      <div
        className="information text"
      >
        <span>
          Last updated: 6th June 2020
        </span>
        <h3>
          What are cookies?
        </h3>
        <p
          className="paragraph"
        >
          A “cookie” is a string of information which assigns you a unique identifier that we store on your computer. Your browser then provides that unique identifier to use each time you submit a query to the Site. We use cookies on the Site to, among other things, keep track of services you have used, record registration information, record your user preferences, keep you logged into the Site, facilitate purchase procedures, and track the pages you visit. Cookies help us understand how the Site is being used and improve your user experience.
        </p>
        <h3>
          What cookies does this site use?
        </h3>
        <ul
          className="listOfCookies"
        >
          <li
            className="cookieDescription"
          >
            This site stores cookie data based on the theme selector if you choose to use it. A cookie is written detailing the theme choice which enables the site to remember your theme selection and preserve that selection when you refresh the page.
          </li>
          <li
            className="cookieDescription"
          >
            This site uses cookies to record that you've acknowledged this cookie policy so that we don't have to show it to you more than we need to.
          </li>
        </ul>
        <h3>
          Your control of cookies
        </h3>
        <p
          className="paragraph"
        >
          This site uses cookies, but you can control these through your browser settings. Most browsers allow you to manage cookies saved on your device – just head to the help section of your browser.
        </p>
        <p
          className="paragraph"
        >
          Most browsers are set to accept cookies by default. However, you can remove or reject cookies in your browser’s settings. Please be aware that such action could affect the availability and functionality of the Site.
        </p>
        <h3>
          Amendments to this cookie policy
        </h3>
        <p
          className="paragraph"
        >
          We reserve the right to make changes to this Cookie Policy at any time and for any reason. We will alert you about any changes by updating the “Last Updated” date of this Cookie Policy. Any changes or modifications will be effective immediately upon posting the updated Cookie Policy on the Site, and you waive the right to receive specific notice of each such change or modification.
        </p>
        <p
          className="paragraph"
        >
          You are encouraged to periodically review this Cookie Policy to stay informed of updates. You will be deemed to have been made aware of, will be subject to, and will be deemed to have accepted the changes in any revised Cookie Policy by your continued use of the Site after the date such revised Cookie Policy is posted.
        </p>
      </div>
    </div>
  </div>
</LearnMoreOverlay>
`;
